src/k8s/pkg/k8sd/types/cluster_config.go

```diff
@@ -49,8 +53,8 @@ type K8sDqlite struct {
 	Port int `yaml:"port,omitempty"`
 }
 
-func DefaultClusterConfig() ClusterConfig {
-	return ClusterConfig{
+func SetClusterConfigDefaults(b *apiv1.BootstrapConfig) (ClusterConfig, error) {
+	config := ClusterConfig{
```

Comment:
This should be a `(*ClusterConfig).SetDefaults() error`, that would look like this:

```go
func (c *ClusterConfig) SetDefaults() {
  if c.Network.PodCIDR == "" {
    c.Network.PodCIDR = "10.1.0.0/16"
  }
}
```

Reply:
Did you want me to keep the merging part of the default config and bootstrap config and then set this default?

Reply:
usage should look like this:

```
// NOTE: must include error checking
clusterConfig := ClusterConfigFromBootstrapConfig(bootstrapConfig)
clusterConfig.SetDefaults()
clusterConfig.Validate()
```

Reply:
Should we use if statements for the other defaults in case we do decide to add configurations to `ClusterConfigFromBootstrapConfig` or keep it like this?
```
func (c *ClusterConfig) SetDefaults() {
	if c.Network.PodCIDR == "" {
		c.Network.PodCIDR = "10.1.0.0/16"
	}
	c.Network.ServiceCIDR = "10.152.183.0/24"
	c.APIServer.Datastore = "k8s-dqlite"
	c.APIServer.SecurePort = 6443
	c.APIServer.AuthorizationMode = "Node,RBAC"
	c.K8sDqlite.Port = 9000
}
```

Reply:
For more than one podCIDR are we expecting `PodCIDR: "10.1.0.0/16, 2001:0db8::/32"` or  no space:`PodCIDR: "10.1.0.0/16,2001:0db8::/32"`?

Reply:
no spaces for the CIDRs. also, yes, let's set these only if the current values are unset, similar to the PodCIDR